ckan/views/__init__.py

```diff
@@ -96,6 +96,28 @@ def set_cors_headers_for_response(response):
     return response
 
 
+def set_cache_control_headers_for_response(response):
+
+    # no_cache should be None when caching is allowed
+    allow_cache = request.environ.get(u'__no_cache__')
+
+    if u'Pragma' in response.headers:
+        del response.headers["Pragma"]
+
+    if allow_cache is None:
```

Comment:
I find this really confusing. I get that if `__no_cache__` is not on `request.environ` we should allow caching but maybe we can define this as
```python
allow_cache = request.environ.get(u'__no_cache__') is None
```
or

```python
allow_cache = u'__no_cache__' not in request.environ
```

so `allow_cache` is actually `True`